test/e2e/operator_test.go

```diff
@@ -566,3 +572,13 @@ func fetchRouterCanonicalHostname(t *testing.T, routeName types.NamespacedName)
 	}
 	return canonicalName, nil
 }
+
+func ingressConditionStatus(ing routev1.RouteIngress, t routev1.RouteIngressConditionType) corev1.ConditionStatus {
```

Comment:
I would change this to `ingressConditionHasStatus(ing routev1.RouteIngress, condition routev1.RouteIngressConditionType, status corev1.ConditionStatus) bool` because right now we are return `corev1.ConditionUnknown` if we cannot find the condition which is strictly not true.